src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -467,49 +394,43 @@ export function fillLoadoutFromEquipped(
     const newEquippedItems = store.items.filter(
       (item) => item.equipped && itemCanBeInLoadout(item) && itemMatchesCategory(item, category)
     );
-    const mods: number[] = [];
+    const modsByBucket: { [bucketHash: number]: number[] } = {};
     for (const item of newEquippedItems) {
       if (!(item.bucket.hash in equippedItemsByBucket)) {
-        const loadoutItem = convertToLoadoutItem(item, true, 1);
-        if (item.bucket.hash === BucketHashes.Subclass) {
-          loadoutItem.socketOverrides = createSocketOverridesFromEquipped(item);
+        loadout = addItem(defs, item, true)(loadout);
+
+        // Only save fashion for the items we added
```

Comment:
This is a behavior change - previously "fill in using equipped" always replaced all the fashion with the equipped items' fashion. Now we only fill it in for empty fashion in the same bucket as added items.